test_filter_outputs/src/main.rs

```diff
@@ -0,0 +1,272 @@
+        .to_vec();
+        for node in &regular_nodes {
+            // node: ID, capacity, egress rate, generation rate, plugin
+            simulator.add_node(node, 10, 5, 0, plugin);
+        }
+        simulator.add_node("loadgenerator-v1", 10, 1, 1, None);
+        simulator.add_storage(STORAGE_NAME);
+
+        // add all connections to storage
+        for node in &regular_nodes {
+            simulator.add_edge(1, node, STORAGE_NAME, true);
+        }
+
+        // src: traffic generator
+        simulator.add_edge(1, "loadgenerator-v1", "productpage-v1", true);
+        // src: product page
+        simulator.add_edge(1, "productpage-v1", "reviews-v1", false);
+        simulator.add_edge(1, "productpage-v1", "reviews-v2", false);
+        simulator.add_edge(1, "productpage-v1", "reviews-v3", false);
+        simulator.add_edge(1, "productpage-v1", "details-v1", true);
+        // src: reviews
+        simulator.add_edge(1, "reviews-v1", "ratings-v1", false);
+        simulator.add_edge(1, "reviews-v2", "ratings-v1", false);
+        simulator.add_edge(1, "reviews-v3", "ratings-v1", false);
+        return simulator;
+    }
+
+    // Count is weird - if you make the nodes have the same names, as makes sense to me, then it doesn't work with c++
+    // but I'm not sure what exactly the query is saying if they have different names - so we'll leave it for now until
+    // we get language semantics ironed out
+    #[test_case(
+        "breadth_histogram",
+        "breadth_histogram.cql",
+        Some("histogram.rs"),
+        "Hist:  (1, 1) \n\nHist:  (1, 2) \n\nHist:  (1, 3) \n\nHist:  (1, 4) \n\nHist:  (1, 5) \n\nHist:  (1, 6) \n\nHist:  (1, 7) \n\n" ; "breadth_histogram_test"
+    )]
+    #[test_case(
+        "height_histogram",
+        "height_histogram.cql",
+        Some("histogram.rs"),
+        "Hist:  (2, 1) \n\nHist:  (2, 2) \n\nHist:  (2, 3) \n\nHist:  (2, 4) \n\nHist:  (2, 5) \n\nHist:  (2, 6) \n\nHist:  (2, 7) \n\n" ; "height_histogram_test"
+    )]
+    #[test_case(
+        "response_code_count",
+        "response_code_count.cql",
+        Some("count.rs"),
+        "1\n2\n3\n4\n5\n6\n7\n" ; "response_code_count_test"
+    )]
+    #[test_case(
+        "response_size_avg",
+        "response_size_avg.cql",
+        Some("avg.rs"),
+        "1\n1\n1\n1\n1\n1\n1\n" ; "response_size_avg_test"
+    )]
+    #[test_case(
+        "return_test",
+        "return.cql",
+        None,
+        "1\n1\n1\n1\n1\n1\n1\n" ; "return_test"
+    )]
+    #[test_case(
+        "return_height",
+        "return_height.cql",
+        None,
+        "2\n2\n2\n2\n2\n2\n2\n" ; "return_height_test"
+    )]
+    fn test(query_id: &str, query_name: &str, udfs: Option<&str>, expected_output: &str) {
+        // 1. get directories
+        let directories_wrapped = get_dirs();
+        assert!(directories_wrapped != None);
+        let directories = directories_wrapped.unwrap();
+
+        let temp_dir_buf = &mut directories["SIM_DIR"].clone();
```

Comment:
The temp dir should be here, in the tracing env. Not in the simulator. The Cargo,toml is provided by the tracing_env with the appropriate paths.